sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc

```diff
@@ -37,155 +37,168 @@
+  auto const yrange = p.get<std::vector<double> > ("YRange");
+  auto const zrange = p.get<std::vector<double> > ("ZRange");
+
+  auto const part_cfg = p.get<fhicl::ParameterSet>("ParticleParameter");
+  _gamma_beta_range = p.get<std::array<double,2> >("GammaBetaRange", {0.0,0.0});
+
+  _param_v.clear();
+  auto const pdg_v      = part_cfg.get<std::vector<std::vector<int>    > > ("PDGCode");
+  auto const minmult_v  = part_cfg.get<std::vector<unsigned short> > ("MinMulti");
+  auto const maxmult_v  = part_cfg.get<std::vector<unsigned short> > ("MaxMulti");
+  auto const weight_v   = part_cfg.get<std::vector<double> > ("ProbWeight");
+
+  auto kerange_v  = part_cfg.get<std::vector<std::vector<double> > > ("KERange");
+  auto momrange_v = part_cfg.get<std::vector<std::vector<double> > > ("MomRange");
+
+  if( (kerange_v.empty() && momrange_v.empty()) ||
+      (!kerange_v.empty() && !momrange_v.empty()) ) {
+    this->abort("Only one of KERange or MomRange must be empty!");
+  }
 
-    bool use_mom = false;
-    if(kerange_v.empty()){
-        kerange_v = momrange_v;
-        use_mom = true;
-    }
-    // sanity check
-    if( pdg_v.size() != kerange_v.size() ||
-       pdg_v.size() != minmult_v.size() ||
-       pdg_v.size() != maxmult_v.size() ||
-       pdg_v.size() != weight_v.size() )
-        this->abort("configuration parameters have incompatible lengths!");
-
-    // further sanity check (1 more depth for some double-array)
-    for(auto const& r : pdg_v    ) { if(              r.empty()  ) this->abort("PDG code not given!");                        }
-    for(auto const& r : kerange_v) { if(              r.size()!=2) this->abort("Incompatible legnth @ KE vector!");           }
-
-    size_t multi_min = 0;
-    for(size_t idx=0; idx<minmult_v.size(); ++idx) {
-        if(minmult_v[idx] > maxmult_v[idx]) this->abort("Particle MinMulti > Particle MaxMulti!");
-        if(minmult_v[idx] > _multi_max) this->abort("Particle MinMulti > overall MultiMax!");
-        multi_min += minmult_v[idx];
-    }
-    _multi_min = std::max(_multi_min, multi_min);
-    if(_multi_max < _multi_min) this->abort("Overall MultiMax <= overall MultiMin!");
+  bool use_mom = false;
+  if(kerange_v.empty()){
+    kerange_v = momrange_v;
+    use_mom = true;
+  }
+  // sanity check
+  if( pdg_v.size() != kerange_v.size() ||
+      pdg_v.size() != minmult_v.size() ||
+      pdg_v.size() != maxmult_v.size() ||
+      pdg_v.size() != weight_v.size() )
+    this->abort("configuration parameters have incompatible lengths!");
+
+  // further sanity check (1 more depth for some double-array)
+  for(auto const& r : pdg_v    ) { if(              r.empty()  ) this->abort("PDG code not given!");                        }
+  for(auto const& r : kerange_v) { if(              r.size()!=2) this->abort("Incompatible legnth @ KE vector!");           }
+  if(_gamma_beta_range[0] > _gamma_beta_range[1]) this->abort("Incompatible boost range!");
```

Comment:
if this is velocity, we should check if both are bounded between 0 and 1

Reply:
`_gamma_beta` denotes gamma times beta, so the possible range is 0 to inf.